services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/UploadWithUnknownContentLengthHelper.java

```diff
@@ -163,9 +163,17 @@ public void onSubscribe(Subscription s) {
 
         @Override
         public void onNext(CloseableAsyncRequestBody asyncRequestBody) {
+            if (asyncRequestBody == null) {
```

Comment:
Do we need to add `isDone = true` here just like with `onError()` ?

Reply:
Do you mean moving `isDone` check before this? Probably not, since we want to always error out if the item received is null

Reply:
Yeah, it's not necessary since we will throw exception now